src/test/java/com/icthh/xm/ms/balance/repository/PostgressConfigIntTest.java

```diff
@@ -0,0 +1,147 @@
+import org.testcontainers.containers.PostgreSQLContainer;
+
+@Slf4j
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = {BalanceApp.class, SecurityBeanOverrideConfiguration.class})
+@ContextConfiguration(initializers = {com.icthh.xm.ms.balance.repository.PostgressConfigIntTest.Initializer.class})
+@ActiveProfiles("pg-test")
+public class PostgressConfigIntTest {
+
+    @ClassRule
+    public static PostgreSQLContainer postgreSQLContainer = new PostgreSQLContainer("postgres:11.1")
+        .withDatabaseName("balance")
+        .withUsername("sa")
+        .withPassword("sa");
+
+    static class Initializer implements ApplicationContextInitializer<ConfigurableApplicationContext> {
+        public void initialize(ConfigurableApplicationContext configurableApplicationContext) {
+            TestPropertyValues.of(
+                "spring.datasource.url=" + postgreSQLContainer.getJdbcUrl(),
+                "spring.datasource.username=" + postgreSQLContainer.getUsername(),
+                "spring.datasource.password=" + postgreSQLContainer.getPassword()
+                                 ).applyTo(configurableApplicationContext.getEnvironment());
+            log.info("spring.datasource.url: {}", postgreSQLContainer.getJdbcUrl());
+        }
+    }
+
+    @Autowired
+    private PocketRepository pocketRepository;
+
+    @Autowired
+    private BalanceRepository balanceRepository;
+
+    @Autowired
+    private TenantContextHolder tenantContextHolder;
+
+    @Autowired
+    private LepManager lepManager;
+
+    @Autowired
+    private XmAuthenticationContextHolder xmAuthenticationContextHolder;
+
+    @Before
+    public void beforeTransaction() {
+        TenantContextUtils.setTenant(tenantContextHolder, "public");
+        lepManager.beginThreadContext(scopedContext -> {
+            scopedContext.setValue(THREAD_CONTEXT_KEY_TENANT_CONTEXT, tenantContextHolder.getContext());
+            scopedContext.setValue(BINDING_KEY_AUTH_CONTEXT, xmAuthenticationContextHolder.getContext());
+        });
+    }
+
+    @After
+    @Override
+    public void finalize() {
+        tenantContextHolder.getPrivilegedContext().destroyCurrentContext();
+        lepManager.endThreadContext();
+    }
+
+
+    @Test
+    public void testSearchUsingMetadata() {
+
+        Balance balance = balanceRepository.save(new Balance().typeKey("BALANCE").key("8").entityId(8L));
+        Pocket falsePocket = pocketRepository.save(pocket(18L, balance, of("bvalue", "false")));
+        Pocket truePocket = pocketRepository.save(pocket(28L, balance, of("bvalue", "true")));
+        Pocket first3Pocket = pocketRepository.save(pocket(38L, balance, of("eId", "3")));
+        Pocket second3Pocket = pocketRepository.save(pocket(48L, balance, of("eId", "3")));
+        pocketRepository.save(pocket(58L, balance, of("eId", "4")));
+        pocketRepository.save(pocket(68L, balance, of("eId", "4")));
+
+        List<Pocket> pockets3 = pocketRepository.findAll(Specification.where(
```

Comment:
Is it possible to filter inside json not on the first level field?
for example if we have a pocket 
```
{
    "metadata_json": {
            "bvalue": true,
            "inner_container": {
                    "inValue": 123
             }
     }
}
```

how can we filter by `inValue`?